### PR TITLE
Move to netcoreapp2.1

### PR DESCRIPTION
**Pull request by jaredpar**
After working with the CoreClr team it turns out part of our CoreClr
flakiness on Linux is due to bugs fixed in netcoreapp2.1. Given that and
that our consumers are all on netcoreapp2.1 already I'm moving us to
2.1.

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @dotnet/roslyn-infrastructure, @tmat for review

File: src/Compilers/Server/VBCSCompilerTests/CompilerServerTests.cs [hunk 1]
```diff
@@ -101,7 +101,7 @@ private static void CheckForBadShared(List<string> arguments)
 
         private static void ReferenceNetstandardDllIfCoreClr(TempDirectory currentDirectory, List<string> arguments)
         {
-#if NETCOREAPP2_0
+#if NETCOREAPP2_1
```
Comment (tmat) [inline src/Compilers/Server/VBCSCompilerTests/CompilerServerTests.cs]: Would it be better to use `!NET46`? We will likely update NetCoreApp target more often than Desktop FX.

Comment (jaredpar) [reply src/Compilers/Server/VBCSCompilerTests/CompilerServerTests.cs]: Yes. I will change.

File: src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_CallerInfoAttributes.cs [hunk 2]
```diff
@@ -1538,10 +1538,15 @@ partial class A { static void Main5() { Log(); } }
...
+            // On CoreClr the '*' is a legal path character
+            // https://github.com/dotnet/docs/issues/4483
+            var expectedStarPath = ExecutionConditionUtil.IsCoreClr
+                ? @"C:\A\B\*"
+                : "*";
+            CompileAndVerify(compilation, expectedOutput: $@"
 1: 'C:\filename'
 2: 'C:\A\B\a\c\d.cs'
-3: '*'
+3: '{expectedStarPath}'
```
Comment (tmat) [inline src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_CallerInfoAttributes.cs]: Allowing `*` in path will break SourceLink. Can we change the compiler to preserve the current behavior when `*` is in the path?

Comment (tmat) [reply src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_CallerInfoAttributes.cs]: Actually, never mind. It's not a problem.

**Comment by jaredpar** [comment]
The win64 leg is going to time out because we got a machine with four cores again. Following up with core-eng. The equivalent leg passed in VSTS though hence feel comfortable merging

**Pull request merged**